streams/cluster/src/main/java/com/hortonworks/streamline/streams/cluster/service/metadata/StormMetadataService.java

```diff
@@ -153,45 +177,49 @@ private Client newHttpClient() {
-
-            if (stormUiComp == null) {
-                throw new ServiceComponentNotFoundException(clusterId, ServiceConfigurations.STORM.name(), ComponentPropertyPattern.STORM_UI_SERVER.name());
-            }
-
+            final Component stormUiComp =
+                    EnvironmentServiceUtil.getComponent(
+                            environmentService, clusterId, AMBARI_JSON_SERVICE_STORM, AMBARI_JSON_COMPONENT_STORM_UI_SERVER);
             return new HostPort(stormUiComp.getHosts().get(0), stormUiComp.getPort());
         }
     }
 
     /**
      * @return List of storm topologies as returned by Storm's REST API
      */
-    public StormTopologies getTopologies() throws IOException {
-        final Map<String, ?> jsonAsMap = JsonClientUtil.getEntity(httpClient.target(tplgySumUrl), Map.class);
-        List<String> topologies = Collections.emptyList();
-        if (jsonAsMap != null) {
-            final List<Map<String, String>> topologiesSummary = (List<Map<String, String>>) jsonAsMap.get(STORM_REST_API_TOPOLOGIES_KEY);
-            if (topologiesSummary != null) {
-                topologies = new ArrayList<>(topologiesSummary.size());
-                for (Map<String, String> tpSum : topologiesSummary) {
-                    topologies.add(tpSum.get(STORM_REST_API_TOPOLOGY_ID_KEY));
+    public StormTopologies getTopologies() throws IOException, PrivilegedActionException {
+        return executeSecure(() -> {
+                    final Map<String, ?> jsonAsMap = JsonClientUtil.getEntity(httpClient.target(tplgySumUrl), Map.class);
+                    List<String> topologies = Collections.emptyList();
+                    if (jsonAsMap != null) {
+                        final List<Map<String, String>> topologiesSummary = 
+                                (List<Map<String, String>>) jsonAsMap.get(STORM_REST_API_TOPOLOGIES_KEY);
+                        if (topologiesSummary != null) {
+                            topologies = new ArrayList<>(topologiesSummary.size());
+                            for (Map<String, String> tpSum : topologiesSummary) {
+                                topologies.add(tpSum.get(STORM_REST_API_TOPOLOGY_ID_KEY));
+                            }
+                        }
+                    }
+                    return new StormTopologies(topologies, getSecurity());
                 }
-            }
-        }
-        return new StormTopologies(topologies, new Security(securityContext, new Authorizer(false), getPrincipals(), getKeytabs()));
+        );
+    }
+
+    private <T, E extends Exception> T executeSecure(SupplierException<T, E> action) throws PrivilegedActionException, E {
+        return action.get();    // For now it executes without security context. Uncomment bellow for security context.
+//        return SecurityUtil.execute(action, securityContext, subject);
```

Comment:
@priyank5485 I left it like this because for my tests in secure (and insecure) mode it is the only way it works. With subject.doAs() oddly it does not work. Perhaps it is related with the comment that @HeartSaVioR made in [here](https://github.com/hortonworks/streamline/pull/596) as a follow up to his [patch](https://github.com/hortonworks/streamline/pull/598)


Is the code with the line currently commented out working for everyone? If so I can revet it to match this [change](https://github.com/hortonworks/streamline/pull/596), and will identify why it does not work for my tests.